File: M1/azor-chatdog-js/src/commands/sessionList.ts
/**
 * Session list command (matches Python version)
 */

import { listSessions } from '../files/sessionFiles.js';
import { printHelp } from '../cli/console.js';

/**
 * Display list of all sessions
 */
export function displaySessionList(): void {
  const sessions = listSessions();

  if (sessions.length === 0) {
    printHelp('\nBrak zapisanych sesji.');
    return;
  }

  printHelp('\n--- Dostępne zapisane sesje (ID) ---');
  for (const session of sessions) {
    printHelp(
      `- ID: ${session.session_id} (Wiadomości: ${session.message_count}, Ost. aktywność: ${session.last_modified.toLocaleString()})`
    );
  }
  printHelp('------------------------------------');
}

File: M1/azor-chatdog-py/src/commands/session_list.py
from files import session_files
from cli import console

def list_sessions_command():
    """Displays a formatted list of available sessions."""
    sessions = session_files.list_sessions()
    if sessions:
        console.print_help("\n--- Dostępne zapisane sesje (ID) ---")
        for session in sessions:
            if session.get('error'):
                console.print_error(f"- ID: {session['id']} ({session['error']})")
            else:
                console.print_help(f"- ID: {session['id']} (Wiadomości: {session['messages_count']}, Ost. aktywność: {session['last_activity']})")
        console.print_help("------------------------------------")
    else:
        console.print_help("\nBrak zapisanych sesji.")
